src/ImageSharp/MetaData/ImageFrameMetaData.cs

```diff
@@ -28,32 +31,57 @@ internal ImageFrameMetaData(ImageFrameMetaData other)
             DebugGuard.NotNull(other, nameof(other));
 
-            this.FrameDelay = other.FrameDelay;
-            this.DisposalMethod = other.DisposalMethod;
+            foreach (KeyValuePair<string, object> meta in other.metaData)
+            {
+                this.metaData.Add(meta.Key, meta.Value);
+            }
         }
 
         /// <summary>
-        /// Gets or sets the frame delay for animated images.
-        /// If not 0, when utilized in Gif animation, this field specifies the number of hundredths (1/100) of a second to
-        /// wait before continuing with the processing of the Data Stream.
-        /// The clock starts ticking immediately after the graphic is rendered.
+        /// Clones this ImageFrameMetaData.
         /// </summary>
-        public int FrameDelay { get; set; }
+        /// <returns>The cloned instance.</returns>
+        public ImageFrameMetaData Clone() => new ImageFrameMetaData(this);
 
         /// <summary>
-        /// Gets or sets the disposal method for animated images.
-        /// Primarily used in Gif animation, this field indicates the way in which the graphic is to
-        /// be treated after being displayed.
+        /// Adds or updates the specified key and value to the <see cref="ImageMetaData"/>.
         /// </summary>
-        public DisposalMethod DisposalMethod { get; set; }
+        /// <param name="key">The key of the metadata to add.</param>
+        /// <param name="value">The value of the element to add.</param>
+        /// <exception cref="ArgumentNullException">key is null.</exception>
+        /// <exception cref="ArgumentNullException">value is null.</exception>
+        /// <exception cref="ArgumentException">An element with the same key already exists in the <see cref="ImageMetaData"/>.</exception>
+        public void AddOrUpdateMetaData(string key, object value)
+        {
+            // Don't think this needs to be threadsafe.
+            Guard.NotNull(value, nameof(value));
+            this.metaData[key] = value;
+        }
 
         /// <summary>
-        /// Clones this ImageFrameMetaData.
+        /// Gets the metadata value associated with the specified key.
         /// </summary>
-        /// <returns>The cloned instance.</returns>
-        public ImageFrameMetaData Clone()
+        /// <typeparam name="T">The type of value.</typeparam>
+        /// <param name="key">The key of the value to get.</param>
+        /// <param name="value">
+        /// When this method returns, contains the metadata value associated with the specified key,
+        /// if the key is found; otherwise, the default value for the type of the value parameter.
+        /// This parameter is passed uninitialized.
+        /// </param>
+        /// <returns>
+        /// true if the <see cref="ImageMetaData"/> contains an element with
+        /// the specified key; otherwise, false.
+        /// </returns>
+        public bool TryGetMetaData<T>(string key, out T value)
```

Comment:
Can't we find a more type-safe key than `string`?

If decide to stick with the one entry/format solution then I would suggest to use `IImageFormat` as a key (after making all formats singletons).

Reply:
I wanted to use `IImageFormat` but what instance do you use as a key and how do you access it from the extension methods?